### PR TITLE
More gracefully handle forms with missing XML

### PR DESCRIPTION
**Pull request by millerdev**
Plus a minor refactor. Review by commit 🐠 

@orangejenny

### CODE REVIEW

File: corehq/apps/couch_sql_migration/management/commands/show_started_migrations.py [hunk 1]
```diff
@@ -30,18 +30,24 @@ def handle(self, show_stats=False, **options):
...
+        ))
+
+    def print_stats(self, item):
+        try:
+            stats = get_diff_stats(item.domain)
+            if stats:
+                stats = format_diff_stats(stats)
+                print("    " + stats.replace("\n", "\n    "))
+        except Exception as err:
+            print("    Cannot get diff stats: {}".format(err))
```
Comment (orangejenny) [inline corehq/apps/couch_sql_migration/management/commands/show_started_migrations.py]: Realize you just moved this code, but doesn't `err` need to be cast with `six.text_type` for python 3?

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 2]
```diff
@@ -765,6 +766,19 @@ def _fix_replacement_form_problem_in_couch(doc):
...
 
+def sql_form_to_json(form):
+    """Serialize SQL form to JSON
+
+    Handles missing form XML gracefully.
+    """
+    try:
+        form.get_xml()
+    except BlobNotFound:
+        form.get_xml.get_cache(form)[()] = ""
```
Comment (orangejenny) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: What does the `[()]` do?

File: corehq/apps/couch_sql_migration/management/commands/show_started_migrations.py (hunk 1 above)
Comment (millerdev) [reply corehq/apps/couch_sql_migration/management/commands/show_started_migrations.py]: Formatting it in a string does that automatically.

File: corehq/apps/couch_sql_migration/couchsqlmigration.py (hunk 2 above)
Comment (millerdev) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: `()` is an empty tuple, which is the cache key used by `@memoized` when a function is called with no arguments. So `[()] = ""` assigns an empty string to the item in the cache (a dict) referenced by the empty tuple.

**Review by orangejenny — APPROVED** [review]
Is it worth renaming `process_stock` to make it clearer that it mutates `case_db`?

**Comment by millerdev** [comment]
> Is it worth renaming `process_stock` to make it clearer that it mutates `case_db`?

Yes, but it's not really in the scope of what I'm doing right now and I don't have head space to go chasing every rabbit trail of things that could be improved about the form processing code. Someday I hope to get around to it—hopefully when I remove the couch form processing code.

**Pull request merged**